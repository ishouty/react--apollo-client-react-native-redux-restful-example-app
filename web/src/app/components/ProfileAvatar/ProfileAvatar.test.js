import React, {Component} from "react";
import {shallow} from "enzyme"
import toJson from "enzyme-to-json"
import ProfileAvatarTest from "./ProfileAvatar"


describe('ProfileAvatar', ()=> {

    test('Component should render match snap shot', () => {

        const component = shallow(<ProfileAvatarTest profile={{ profileImage: "profileImage.jpg" }} />)
        const tree = toJson(component)
        expect(tree).toMatchSnapshot()

    })

    test('Component className should be profile-avatar-medium by default ', () => {

        const component = shallow(<ProfileAvatarTest profile={{ profileImage: "profileImage.jpg" }} />)
        expect(component.find('.profile-avatar-medium')).toHaveLength(1)

    })

    test('Component className should be profile-avatar-large when passing prop type size = large ', () => {

        const component = shallow(<ProfileAvatarTest profile={{ profileImage: "profileImage.jpg" }} size="large" />)
        expect(component.find('.profile-avatar-large')).toHaveLength(1)

    })

    test('Component ProfileAvatar onclick should call callback function', () => {

        const mockFunction = jest.fn()

        const component = shallow(<ProfileAvatarTest profile={{ profileImage: "profileImage.jpg" }} callback={mockFunction} />)

        //trigger the click event
        component.find('.profile-link').simulate('click')

        expect(mockFunction).toBeCalled()

    })


})



