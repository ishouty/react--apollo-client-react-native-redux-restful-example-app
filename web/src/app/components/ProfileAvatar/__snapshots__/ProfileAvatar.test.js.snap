// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileAvatar Component should render match snap shot 1`] = `
<div
  className="inline"
>
  <div
    className="profile-avatar-medium"
    style={
      Object {
        "backgroundImage": "url(profileImage.jpg)",
      }
    }
  />
</div>
`;
